graylog2-web-interface/src/components/indices/IndexSetFieldTypes/IndexSetCustomFieldTypeRemoveModal.tsx

```diff
@@ -68,11 +71,51 @@ const indexSetsStoreMapper = ({ indexSets }: IndexSetsStoreState): Record<string
 const IndexSetCustomFieldTypeRemoveContent = ({ fields, indexSets, setRotated, rotated, indexSetIds }: ContentProps) => {
   const fieldsStr = fields.join(', ');
   const indexSetsStr = indexSetIds.map((id) => indexSets[id].title).join(', ');
+  const { customFieldMappingsByField, name: profileName, id: profileId } = useIndexProfileWithMappingsByField();
+  const { overriddenIndexFieldsStr, overriddenProfilesFieldsWithType } = useMemo<{ overriddenIndexFieldsStr: string, overriddenProfilesFieldsWithType: Array<{ field: string, type: string }> }>(() => {
+    const { overriddenIndexFields, overriddenProfilesFields } = fields.reduce((acc, cur) => {
+      if (customFieldMappingsByField[cur]) {
+        acc.overriddenProfilesFields.push({ field: cur, type: customFieldMappingsByField[cur] });
+      } else {
+        acc.overriddenIndexFields.push(cur);
+      }
+
+      return acc;
+    }, { overriddenIndexFields: [], overriddenProfilesFields: [] });
+
+    return ({
+      overriddenIndexFieldsStr: overriddenIndexFields.join(', '),
+      overriddenProfilesFieldsWithType: overriddenProfilesFields,
+    });
+  }, [customFieldMappingsByField, fields]);
 
   return (
     <div>
       <Alert>
-        After removing the custom field type for <b>{fieldsStr}</b> in <b>{indexSetsStr}</b> the settings of your search engine will be used
+        After removing the overridden field type for <b>{fieldsStr}</b> in <b>{indexSetsStr}</b>
+        {overriddenIndexFieldsStr && (
+        <> the
+          settings of your <i>search engine</i> will be applied for
+          fields: <b>{overriddenIndexFieldsStr}</b>.
+        </>
+        )}
+        {!!overriddenProfilesFieldsWithType.length && (
+        <> The settings
+          from <Link to={Routes.SYSTEM.INDICES.FIELD_TYPE_PROFILES.edit(profileId)}>{profileName}</Link> (
+          namely {overriddenProfilesFieldsWithType.map(({ field, type }, index) => {
+          const isLast = index === overriddenProfilesFieldsWithType.length - 1;
+
+          return (
+            <span key={`${field}-${type}`}>
+              <b>{field}:</b> <i>{type}</i>{isLast ? '' : ', '}
+            </span>
+          );
+        })}
+          )
+          {' '}
+          will be applied.
+        </>
+        )}
```

Comment:
Could you please improve the indentation in these cases? For example
```
{!!overriddenProfilesFieldsWithType.length && (
  <>
    The settings from <Link to={Routes.SYSTEM.INDICES.FIELD_TYPE_PROFILES.edit(profileId)}>{profileName}</Link> (namely
    {overriddenProfilesFieldsWithType.map(({ field, type }, index) => {
      const isLast = index === overriddenProfilesFieldsWithType.length - 1;

      return (
        <span key={`${field}-${type}`}>
          <b>{field}:</b> <i>{type}</i>{isLast ? '' : ', '}
        </span>
      );
    })})
    {' '}will be applied.
  </>
)}
```

would be so much easier to read.

Reply:
it would be even easier to read, when we extract the `overriddenProfilesFieldsWithType.map` part in a component.